File: mobile/toastlabplus_app/lib/models/meeting.dart
/// Meeting model for ToastLabPlus
class Meeting {
  final int id;
  final int? clubId;
  final int? meetingNumber;
  final String? theme;
  final DateTime meetingDate;
  final String? startTime;
  final String? endTime;
  final String? location;
  final String status;
  final int? speakerCount;
  final String? notes;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  Meeting({
    required this.id,
    this.clubId,
    this.meetingNumber,
    this.theme,
    required this.meetingDate,
    this.startTime,
    this.endTime,
    this.location,
    required this.status,
    this.speakerCount,
    this.notes,
    this.createdAt,
    this.updatedAt,
  });

  factory Meeting.fromJson(Map<String, dynamic> json) {
    return Meeting(
      id: json['id'] as int,
      clubId: json['clubId'] as int?,
      meetingNumber: json['meetingNumber'] as int?,
      theme: json['theme'] as String?,
      meetingDate: json['meetingDate'] != null
          ? DateTime.parse(json['meetingDate'] as String)
          : DateTime.now(),
      startTime: json['startTime'] as String?,
      endTime: json['endTime'] as String?,
      location: json['location'] as String?,
      status: json['status'] as String? ?? 'DRAFT',
      speakerCount: json['speakerCount'] as int?,
      notes: json['notes'] as String?,
      createdAt: json['createdAt'] != null
          ? DateTime.tryParse(json['createdAt'] as String)
          : null,
      updatedAt: json['updatedAt'] != null
          ? DateTime.tryParse(json['updatedAt'] as String)
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      if (clubId != null) 'clubId': clubId,
      if (meetingNumber != null) 'meetingNumber': meetingNumber,
      if (theme != null) 'theme': theme,
      'meetingDate': meetingDate.toIso8601String().split('T')[0],
      if (startTime != null) 'startTime': startTime,
      if (endTime != null) 'endTime': endTime,
      if (location != null) 'location': location,
      'status': status,
      if (speakerCount != null) 'speakerCount': speakerCount,
      if (notes != null) 'notes': notes,
    };
  }

  String get statusDisplay {
    switch (status) {
      case 'DRAFT':
        return 'Draft';
      case 'OPEN':
        return 'Open for Sign-up';
      case 'CLOSED':
        return 'Closed';
      case 'FINALIZED':
        return 'Finalized';
      case 'COMPLETED':
        return 'Completed';
      case 'CANCELLED':
        return 'Cancelled';
      default:
        return status;
    }
  }
}

File: backend/mcp-server/src/main/java/com/toastlabplus/entity/Meeting.java
package com.toastlabplus.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;

@Entity
@Table(name = "meeting")
public class Meeting {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "club_id", nullable = false)
    @JsonIgnoreProperties({ "hibernateLazyInitializer", "handler" })
    private Club club;

    @Column(length = 200)
    private String title;

    @Column(length = 200)
    private String theme;

    @Column(name = "meeting_date", nullable = false)
    private LocalDate meetingDate;

    @Column(name = "start_time", nullable = false)
    private LocalTime startTime;

    @Column(name = "end_time")
    private LocalTime endTime;

    @Column(length = 200)
    private String location;

    @Column(nullable = false, length = 20)
    private String status = "DRAFT";

    @Column(name = "meeting_number")
    private Integer meetingNumber;

    @Column(name = "speaker_count")
    private Integer speakerCount = 3;

    @Column(name = "template_id")
    private Long templateId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "schedule_id")
    @JsonIgnoreProperties({ "hibernateLazyInitializer", "handler" })
    private MeetingSchedule schedule;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by")
    @JsonIgnoreProperties({ "hibernateLazyInitializer", "handler" })
    private Member createdBy;

    @Column(name = "created_at")
    private LocalDateTime createdAt = LocalDateTime.now();

    @Column(name = "updated_at")
    private LocalDateTime updatedAt = LocalDateTime.now();

    // Status constants
    public static final String STATUS_DRAFT = "DRAFT";
    public static final String STATUS_OPEN = "OPEN";
    public static final String STATUS_CLOSED = "CLOSED";
    public static final String STATUS_FINALIZED = "FINALIZED";
    public static final String STATUS_COMPLETED = "COMPLETED";
    public static final String STATUS_CANCELLED = "CANCELLED";

    // Constructors
    public Meeting() {
    }

    // Getters and Setters
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Club getClub() {
        return club;
    }

    public void setClub(Club club) {
        this.club = club;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getTheme() {
        return theme;
    }

    public void setTheme(String theme) {
        this.theme = theme;
    }

    public LocalDate getMeetingDate() {
        return meetingDate;
    }

    public void setMeetingDate(LocalDate meetingDate) {
        this.meetingDate = meetingDate;
    }

    public LocalTime getStartTime() {
        return startTime;
    }

    public void setStartTime(LocalTime startTime) {
        this.startTime = startTime;
    }

    public LocalTime getEndTime() {
        return endTime;
    }

    public void setEndTime(LocalTime endTime) {
        this.endTime = endTime;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public Member getCreatedBy() {
        return createdBy;
    }

    public void setCreatedBy(Member createdBy) {
        this.createdBy = createdBy;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public LocalDateTime getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(LocalDateTime updatedAt) {
        this.updatedAt = updatedAt;
    }

    public Integer getMeetingNumber() {
        return meetingNumber;
    }

    public void setMeetingNumber(Integer meetingNumber) {
        this.meetingNumber = meetingNumber;
    }

    public Integer getSpeakerCount() {
        return speakerCount;
    }

    public void setSpeakerCount(Integer speakerCount) {
        this.speakerCount = speakerCount;
    }

    public Long getTemplateId() {
        return templateId;
    }

    public void setTemplateId(Long templateId) {
        this.templateId = templateId;
    }

    public MeetingSchedule getSchedule() {
        return schedule;
    }

    public void setSchedule(MeetingSchedule schedule) {
        this.schedule = schedule;
    }
}
